sample_client.py

```diff
@@ -0,0 +1,176 @@
+import logging
+import grpc
+import os
+
+import fire
+from google.protobuf.struct_pb2 import Struct
+from docker import auth
+
+import manager_pb2_grpc as pb2_grpc
+import manager_pb2 as pb2
+
+logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.DEBUG)
+
+attributes_by_pattern = {
+    "*.jpg": {
+        "images": True,
+    },
+    "*.json": {
+        "metadata": True,
+    },
+}
+
+
+def create_collection() -> pb2.Collection:
+    """
+    Create a collection configuration for content publishing.
+    """
+    files_list = []
+    for pattern, attr in attributes_by_pattern.items():
+        attributes_struct = Struct()
+        attributes_struct.update(attr)
+
+        for key, value in attr.items():
+            logging.info(
+                "Applying attribute %s=%s to file pattern %s", key, value, pattern
+            )
+
+        file = pb2.File(file=pattern, attributes=attributes_struct)
+        files_list.append(file)
+
+    return pb2.Collection(files=files_list)
+
+
+def create_auth_config(reference: str) -> pb2.AuthConfig:
+    """
+    Create an authentication configuration based on given
+    reference and default configuration locations.
+    This loads from ~/.docker/config.json.
+    """
+    auth_configs = auth.load_config()
+    registry, _ = auth.resolve_repository_name(reference)
+    creds = auth.resolve_authconfig(auth_configs, registry)
+    if creds is not None:
+        auth_config = pb2.AuthConfig(
+            server_address=f"https://{registry}",
```

Comment:
there is no guarantee it will use `https`

Reply:
This is a good point. The protocol isn't used aside from the fact it satisfies the requirement to pass a server address and not a registry host. Perhaps a better solution is to change the gRPC field from `server address to registry host` and keep the protocol information out. This is only used for host comparison at this point.

Reply:
Proposed change -> https://github.com/uor-framework/uor-client-go/pull/110. The tradeoff is that the client would have to parse the reference to get the registry information. However, it seems like that information is already available, and this change would remove unexpected behavior by setting a protocol that isn't used.